### DISCUSSION TITLE
Assigning values with string/titles as keys

### DISCUSSION THREAD

**Discussion by EllieMacron** [pydantic/pydantic#2428] [Question]
Hi all, I am a new user of `pydantic`. It's nice to see how active the community is 😀 

I'd like to ask if there's a way to populate a model by string as keys.. preferably connected with the type title or something.

Let's say I have this JSON:
```json
{
    "First name": "John",
    "Last name": "Doe"
}
```

and I have this class:

```python
class Person(BaseModel):
    first: str = Field(..., title="First name")
    last: str = Field(..., title="Last name")
```

Aside from mapping the title to the respective field name, what could be my approach in populating the model? "Slugifying" the key is not an option tho.. 

Also, is there a batch setting of attributes onto an existing model? Something similar with `traits`, like `model.trait_set(**traits)`.

Thanks in advance!
I tried searching for some leads about my use cases, but I can't find any. Please feel free to point me to it otherwise

**Comment by PrettyWood** [comment]
Hi @EllieMacron 
I think you're looking for `alias` ;)
```py
class Person(BaseModel):
    first: str = Field(..., alias="First name")
    last: str = Field(..., alias="Last name")

data = {
    "First name": "John",
    "Last name": "Doe"
}

assert repr(Person.parse_obj(data)) == "Person(first='John', last='Doe')"
```

**Reply by ssbarnea** [reply]
It seems that I have a problem related to this one. In [my case](https://github.com/ansible-community/schemas/blob/main/src/ansibleschemas/tasks.py#L123-L126) I am using Field() to dynamically create attributes but I am unable to make use of `title` for them.

```
# a loop:
        kwargs[module_sanitized] = (
            Optional[Mapping[str, Any]],
            Field(title=title, alias=module, description=description
        )

kwargs['__base__'] = TaskModel  # type: ignore
BeremothTaskModel = create_model("BeremothTaskModel", **kwargs)  # type: ignore
```

What I observed is that while description is recorded, the title is fully ignored, mentioning it or not does not have any effect on the generated schema. Alias was used because some of my properties were python reserved keywords (like "copy").

What should I do in order to assure that the exported schema would include a "title" too?
